draft-ietf-taps-interface.md

```diff
@@ -1594,114 +1309,887 @@ does provide the following guarantees about the ordering of operations:
-the Post Sockets interface, from which this work has evolved.
+Each Transport Property takes a value of a property-specific type.
 
---- back
+### Boolean {#transport-props-bool}
 
-# Additional Properties {#appendix-non-consensus}
+A boolean is a data type that can be either "true" or "false". 
+Boolean transport properties should only be used for properties that can not
+be used in an optional way or to query the state of the transport implementation. 
+For optional features, especially in Selection Properties, the usage of the
+Preference type (see {{transport-props-preference}}) is preferred.
 
-The interface specified by this document represents the minimal common interface
-to an endpoint in the transport services architecture {{I-D.ietf-taps-arch}}, based upon
-that architecture and on the minimal set of transport service features
-elaborated in {{I-D.ietf-taps-minset}}. However, the interface has been designed with
-extension points to allow the implementation of features beyond those in the
+### Enumeration {#transport-props-enum}
+
+Enumeration types are used for transport properties that can take one value
+out of a limited set of choices. 
+The representation is implementation dependent.
+
+### Integer {#transport-props-int}
+
+Integer types are used to represent integer numbers. 
+The representation is implementation dependent.
+
+### Preference {#transport-props-preference}
+
+The Preference type is used in most Selection properties on a Preconnection object to constrain Path Selection and Protocol Selection.
+It is a specific instance of the "Enum" type and has five different preference levels:
+
+   | Preference | Effect                                                             |
+   |------------|--------------------------------------------------------------------|
+   | Require    | Select only protocols/paths providing the property, fail otherwise |
+   | Prefer     | Prefer protocols/paths providing the property, proceed otherwise   |
+   | Ignore     | Cancel any default preference for this property                    |
+   | Avoid      | Prefer protocols/paths not providing the property, proceed otherwise |
+   | Prohibit   | Select only protocols/paths not providing the property, fail otherwise |
+
+
+When used on a Connection, this type becomes a (read-only) Boolean representing whether the selected transport supports the requested feature.
+
+
+## Transport Property Classification {#transport-props-classes}
+
+Note:
+: This section is subject to WG discussion on ITEF-102.
+
+Transport Properties – not matter whether they apply to connections, preconnections, or messages – differ in the way they effect the transport system and protocols exposed through the transport system.
+This classification emphasizes two aspects of what and how properties
+effect the transport system in order to correctly set their expectation
+towards these properties:
```

Comment:
In general, this sentence is quite confusing. Perhaps:

"Properties are classified based on how they take effect in the system based on application preferences. There are two factors that determine the classification of a property:"

Reply:
I reworded the second sentence, trying to convey the meaning that there may be multiple aspects in which properties differ, and here we emphasize two of these aspects, so applications know what to expect.